### ISSUE TITLE
Issue with subtraction between array and scalar with the NF

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10029]
Please check the `Buildings.Applications.BaseClasses.Equipment.Validation.PumpParallel` model of the Buildings library. With [1.20.0](https://libraries.openmodelica.org/branches/v1.20/Buildings_latest/files/Buildings_latest_Buildings.Applications.BaseClasses.Equipment.Validation.PumpParallel.err) it works fine, with [master](https://libraries.openmodelica.org/branches/master/Buildings_latest/files/Buildings_latest_Buildings.Applications.BaseClasses.Equipment.Validation.PumpParallel.err) it fails with:
```
[OpenModelica/OMCompiler/Compiler/NFFrontEnd/NFCeval.mo:1062:9-1062:67:writable]
Error: Internal error NFCeval.evalBinarySub failed to evaluate ‘{2, 2} - 1‘
```
The same issue affects other models in the Buildings library.

**Comment by casella** [comment]
Apparently, one of the issues mentioned in [this regression report](https://libraries.openmodelica.org/branches/history/master/2022-12-16%2012:15:59..2022-12-16%2022:26:10.html) is responsible.

**Comment by casella** [comment]
From today's devmeeting discussion, it seems that one of @kabdelhak's commits in the regression report is responsible for the issue.

@perost, I'm still not 100% sure about that. The issue seems to me to be taking place during `NFSimplifyModel.simplify`; besides, @kabdelhak's commits concern the new backend, which is not used here. Can you please elaborate on that here?

**Comment by perost** [comment]
> @perost, I'm still not 100% sure about that. The issue seems to me to be taking place during `NFSimplifyModel.simplify`; besides, @kabdelhak's commits concern the new backend, which is not used here. Can you please elaborate on that here?

The issue is #9953, where he made some changes to NFSimplifyExp in the frontend. This is causing some component references to get incorrect types when simplifying expressions in the frontend.

**Comment by casella** [comment]
Fixed by #10058.
